invisible_cities/cities/psf_liquidata.py

```diff
@@ -0,0 +1,193 @@
+from functools import partial
+
+import numpy  as np
+import tables as tb
+import pandas as pd
+
+from invisible_cities.reco.psf_functions import create_psf
+from invisible_cities.reco.psf_functions import hdst_psf_processing
+
+from liquidata import flat
+from liquidata import get
+from liquidata import put
+from liquidata import pipe
+from liquidata import out
+
+from invisible_cities.io  .kdst_io        import psf_writer
+from invisible_cities.core.core_functions import in_range
+from invisible_cities.core                import system_of_units as units
+from invisible_cities.cities.components   import city
+
+from invisible_cities.database.load_db import DataSiPM
+
+from typing import Sequence
+
+
+# TODO: @Aretno to decide the name of this city
+# TODO: Add city description and etymology
+@city
+def create_psf_city( files_in, file_out, compression, event_range
+                   , detector_db, run_number
+                   , xrange      : Sequence[float]
+                   , yrange      : Sequence[float]
+                   , zbins       : Sequence[float]
+                   , xsectors    : Sequence[float]
+                   , ysectors    : Sequence[float]
+                   , bin_size_xy : float           = 1 * units.mm
+                   ):
+    sipms   = DataSiPM(detector_db, run_number)
+    ranges  = xrange  , yrange
+    sectors = xsectors, ysectors
+
+    nbin      = (np.diff(ranges, axis=1) / bin_size_xy).astype(int).flatten()
+    bin_edges = [np.linspace(*limits, n+1) for limits, n in zip(ranges, nbin)]
+
+    columns_to_drop     = "Xrms Yrms Qc Ec".split()
+    drop_unused_columns = get.dst    *      partial(column_dropper, columns_to_drop)   >> put.dst
+    create_slices_z     = get.dst    * flat(partial(split_in_z, zbins, ranges, sipms)) >> put.z.dst_z
+    create_sectors_xy   = get.dst_z  * flat(partial(split_in_xy, sectors, bin_edges))  >> put.x.y.dst_xy
+    compute_psf         = get.dst_xy *      partial(psf_builder, bin_edges)            >> put.factors.entries.centers
+    build_dataframe     = get.x.y.z.factors.entries.centers * df_builder               >> put.psf
+
+    pipeline = pipe( drop_unused_columns
+                   , create_slices_z
+                   , create_sectors_xy
+                   , compute_psf
+                   , build_dataframe
+                   , get.psf
+                   , out(combine_psfs, None)
+                   )
+
+    result = pipeline(dst_from_files(files_in, "RECO", "Events"))
```

Comment:
Perhaps a name more descriptive than `pipeline` might be good here. OTOH, given that this is the first time many people will ever see a `liquidata` pipeline, emphasizing that it's a pipeline is a good idea. But then `pipe` already does that to a sufficient extent?

Reply:
Since the city will eventually have a more flowery (aka cryptic) name, I would opt to rename `pipeline` such that it really indicates what is meant to be done here.